### PR TITLE
change GenerateFRCMethodNum to GenerateMethodNum 

### PR DESCRIPTION
**Pull request by jennijuju**
change GenerateFRCMethodNum to GenerateMethodNum given they can use FIP-ed metthod(exported builitin actor api) or any method, doesnnt have to be FRC-ed.

### CODE REVIEW

**Review by arajasek — COMMENTED** [review]
I think the goal in this naming was to specify that the method number "generation" happens according to the FRC-0042 spec.

Clearly that wasn't evident given your confusion, but I do want the name to indicate that in some way. "GenerateMethodNum" is too universal IMO.

We could:
- Put frc_0042.go into its own package, so that callsites say `frc0042.GenerateMethodNum`
- Rename to something else that indicates `GenerateMethodNumAsPerTheSpecInFRC_0042`
- Something else :P

**Review by arajasek — APPROVED** [review]
fine

File: builtin/frc0042/frc_0042.go [hunk 1]
```diff
@@ -11,14 +12,14 @@ import (
...
-func GenerateFRCMethodNum(name string) (abi.MethodNum, error) {
+func GenerateMethodNum(name string) (abi.MethodNum, error) {
 	err := validateMethodName(name)
 	if err != nil {
 		return 0, err
 	}
 
 	if name == "Constructor" {
-		return MethodConstructor, nil
+		return builtin.MethodConstructor, nil
```
[the remark this answers is not in the document]
Comment (jennijuju) [reply builtin/frc0042/frc_0042.go]: @shrenujbansal tell us why or approve this PR plz

**Comment by shrenujbansal** [comment]
I don't think I agree with having a whole new package for a single function, one who's use is primarily defined in a single other package. We shouldn't be extremely liberal in adding new packages, because its adds further dependencies to projects and causes build issues like import cycles which become a headache to solve

**Pull request closed without merging**